### PR TITLE
chore: upgrade eslint

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Reference: https://eslint.org/docs/latest/use/configure/migration-guide

1. Updated `eslint` npm package and all plugins
2. Ran converter from https://nx.dev/recipes/tips-n-tricks/flat-config
3. Removed `.eslintrc.json` from apps and libs
4. Converted `eslint-common.json` into `eslint.config.js`
5. Ran `yarn lint --fix`
6. Fixed `vite.config.ts` in all libs

I almost not touched app files, most of them were fixed by `yarn lint --fix`.
Those I've touched manually I've commented in this PR

### CODE REVIEW

File: apps/explorer/src/hooks/usePopper.ts [hunk 1]
```diff
@@ -176,12 +175,11 @@ interface PopperOnClickResult<T extends HTMLElement> {
...
 export const usePopperOnClick = <T extends HTMLElement>(
   placement: Placement = 'top',
-  offset?: number,
+  offset?: number
 ): PopperOnClickResult<T> => {
   const config = usePlacementAndOffset({ placement, offset })
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const { target, show, hide, toggle, ...tooltipProps } = usePopper<T>(config)
+  const { target, show: _show, hide, toggle, ...tooltipProps } = usePopper<T>(config)
```
Comment (shoom3301) [inline apps/explorer/src/hooks/usePopper.ts:182]: Fixed manually

File: apps/cowswap-frontend/src/modules/limitOrders/state/limitOrdersRawStateAtom.ts [hunk 2]
```diff
@@ -20,10 +20,7 @@ export interface LimitOrdersRawState extends ExtendedTradeRawState {
   readonly isUnlocked: boolean
 }
 
-export function getDefaultLimitOrdersState(
-  chainId: SupportedChainId | null,
-  isUnlocked: boolean = false
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/limitOrders/state/limitOrdersRawStateAtom.ts:25]: Fixed manually

**Comment by alfetopito** [comment]
2 unit tests failing 
image

**Review by alfetopito — APPROVED** [review]
Awesome!

Just take a look at the unit tests.

**Review by anxolin — COMMENTED** [review]
Approve

**Pull request merged**